StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1507CodeFixProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace StyleCop.Analyzers.LayoutRules
+{
+    using System.Collections.Immutable;
+    using System.Composition;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.Text;
+    
+    /// <summary>
+    /// Implements a code fix for <see cref="SA1507CodeMustNotContainMultipleBlankLinesInARow"/>.
+    /// </summary>
+    [ExportCodeFixProvider(nameof(SA1507CodeFixProvider), LanguageNames.CSharp)]
+    [Shared]
+    public class SA1507CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1507CodeMustNotContainMultipleBlankLinesInARow.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create("Remove multiple blank lines", token => GetTransformedDocument(context.Document, diagnostic, token)), diagnostic);
+            }
+
+            return Task.FromResult(true);
+        }
+
+        private static async Task<Document> GetTransformedDocument(Document document, Diagnostic diagnostic, CancellationToken token)
+        {
+            var sourceText = await document.GetTextAsync(token).ConfigureAwait(false);
+
+            var startIndex = sourceText.Lines.IndexOf(diagnostic.Location.SourceSpan.Start);
+            int endIndex = startIndex;
+
+            for (var i = startIndex + 1; i < sourceText.Lines.Count; i++)
+            {
+                if (!string.IsNullOrWhiteSpace(sourceText.Lines[i].ToString()))
+                {
+                    endIndex = i - 1;
+                    break;
+                }
+            }
+
+            if (endIndex >= (startIndex + 1))
+            {
+                var replaceSpan = TextSpan.FromBounds(sourceText.Lines[startIndex + 1].SpanIncludingLineBreak.Start, sourceText.Lines[endIndex].SpanIncludingLineBreak.End);
+                var newSourceText = sourceText.Replace(replaceSpan, string.Empty);
+                return document.WithText(newSourceText);
```

Comment:
I looked into the source and it looks like is is only reparsing the changed parts. So this is okay